app/commands/track/update_build_status.rb

```diff
@@ -0,0 +1,359 @@
+class Track::UpdateBuildStatus
+  include Mandate
+
+  initialize_with :track
+
+  def call
+    track.build_status = build_status
+  end
+
+  private
+  def build_status
+    {
+      health:,
+      volunteers:,
+      students:,
+      submissions:,
+      mentor_discussions:,
+      syllabus:,
+      practice_exercises:,
+      test_runner:,
+      representer:,
+      analyzer:
+    }
+  end
+
+  def health
+    num_exemplar = component_health_statuses[:exemplar].to_i
+    num_healthy = component_health_statuses[:healthy].to_i
+
+    return :exemplar if num_exemplar == 5
+    return :healthy if num_exemplar + num_healthy == 5
+
+    :needs_attention
+  end
+
+  memoize
+  def component_health_statuses = [
+    analyzer_health,
+    test_runner_health,
+    representer_health,
+    syllabus_health,
+    practice_exercises_health
+  ].tally
+
+  def volunteers
+    volunteer_user_ids = User::ReputationPeriod.where(
+      period: :forever,
+      about: :track,
+      track_id: track.id
+    ).select(:user_id)
+    volunteers = User.with_attached_avatar.where(id: volunteer_user_ids).order(reputation: :desc).take(12)
+
+    {
+      num_volunteers: volunteer_user_ids.distinct.count,
+      users: SerializeAuthorOrContributors.(volunteers)
+    }
+  end
+
+  def students
+    {
+      num_students: track.num_students,
+      num_students_per_day: average_number_per_day(track.user_tracks, UserTrack)
+    }
+  end
+
+  def submissions
+    {
+      num_submissions: track.submissions.count,
+      num_submissions_per_day: average_number_per_day(track.submissions, Submission)
+    }
+  end
+
+  def mentor_discussions
+    {
+      num_discussions: track.mentor_discussions.count
+    }
+  end
+
+  def test_runner
+    status_counts = track.submissions.where(tests_status: %i[passed failed errored exceptioned]).group(:tests_status).count
```

Comment:
This will probably be quicker without the `where` and then just exclude `pending` from the totalling bit.